File: bukkit/src/main/java/io/wdsj/asw/bukkit/util/message/MessageUtils.java
package io.wdsj.asw.bukkit.util.message;

import ch.jalu.configme.properties.Property;
import org.bukkit.ChatColor;
import org.bukkit.command.CommandSender;

import static io.wdsj.asw.bukkit.AdvancedSensitiveWords.messagesManager;

public class MessageUtils {
    private MessageUtils() {
    }

    public static final char AMPERSAND_CHAR = '&';

    public static String retrieveMessage(Property<String> property) {
        return ChatColor.translateAlternateColorCodes(AMPERSAND_CHAR, messagesManager.getProperty(property));
    }

    public static void sendMessage(CommandSender sender, Property<String> property) {
        String msg = retrieveMessage(property);
        if (!msg.isEmpty()) {
            sender.sendMessage(msg);
        }
    }

    public static void sendMessage(CommandSender sender, String message) {
        if (!message.isEmpty()) {
            sender.sendMessage(ChatColor.translateAlternateColorCodes(AMPERSAND_CHAR, message));
        }
    }
}

File: bukkit/src/main/kotlin/io/wdsj/asw/bukkit/util/message/MessageUtils.kt
package io.wdsj.asw.bukkit.util.message

import ch.jalu.configme.properties.Property
import io.wdsj.asw.bukkit.AdvancedSensitiveWords
import org.bukkit.ChatColor
import org.bukkit.command.CommandSender

object MessageUtils {
    private const val AMPERSAND_CHAR: Char = '&'

    @JvmStatic
    fun retrieveMessage(property: Property<String>): String {
        return ChatColor.translateAlternateColorCodes(
            AMPERSAND_CHAR,
            AdvancedSensitiveWords.messagesManager.getProperty(property)
        )
    }

    @JvmStatic
    fun sendMessage(sender: CommandSender, property: Property<String>) {
        val msg = retrieveMessage(property)
        if (msg.isNotEmpty()) {
            sender.sendMessage(msg)
        }
    }

    @JvmStatic
    fun sendMessage(sender: CommandSender, message: String) {
        if (message.isNotEmpty()) {
            sender.sendMessage(ChatColor.translateAlternateColorCodes(AMPERSAND_CHAR, message))
        }
    }
}
